### PR TITLE
chore(ui): remove react-copy-to-clipboard and add custom hook for copy text

### PR DESCRIPTION
**Pull request by Sachin-chaurasiya**
### Describe your changes :

I worked on removing `react-copy-to-clipboard` and added a custom hook for copy-to-clipboard functionality with native clipboard API.

image

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/59080942/224564466-207abafa-b99d-47e5-8f9a-ab355dc0fbc7.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/hooks/useClipBoard.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  callBack?: () => void
+) => {
+  // local state
+  const [hasCopied, setHasCopied] = useState(false);
+  const [valueState, setValueState] = useState(value);
+
+  // handlers
+  const handleCopy = useCallback(async () => {
+    try {
+      await navigator.clipboard.writeText(valueState);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/hooks/useClipBoard.ts:34]: Can you verify the compatibility of this feature in other browsers?
https://caniuse.com/?search=clipboard

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/hooks/useClipBoard.ts:34]: @chirag-madlani , Yes, I did and it has good support for all major browsers.

File: openmetadata-ui/src/main/resources/ui/src/components/buttons/CopyToClipboardButton/CopyToClipboardButton.tsx [hunk 2]
```diff
@@ -33,44 +31,32 @@ export const CopyToClipboardButton: FunctionComponent<Props> = ({
...
-  const [copied, setCopied] = useState<boolean>(false);
-
-  const handleCopying = () => {
-    setCopied(true);
-    onCopy?.();
-    setTimeout(() => {
-      setCopied(false);
-    }, copyTimer);
-  };
+  const { hasCopied, onCopyToClipBoard } = useClipboard(
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/buttons/CopyToClipboardButton/CopyToClipboardButton.tsx:34]: Can we improve unit tests for this component?
Interaction and negative cases?

1. Callback should have been called
2. After timeout tooltip should be hidden
3. Clipboard must have text once hit copy
4. What happen if we don't have access to navigator APIs?

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/buttons/CopyToClipboardButton/CopyToClipboardButton.tsx:34]: Sure @chirag-madlani , I will work on it.

Comment (Sachin-chaurasiya) [reply openmetadata-ui/src/main/resources/ui/src/components/buttons/CopyToClipboardButton/CopyToClipboardButton.tsx:34]: @chirag-madlani , I have pushed the changes, please review.

**Review by chirag-madlani — APPROVED** [review]
Thanks @Sachin-chaurasiya

**Pull request merged**